### ISSUE TITLE
Unable to delete files on Windows without a Recycle Bin

### ISSUE THREAD

**Issue by manuelbieh** [atom/tree-view#856]
The issue is probably related to https://github.com/atom/tree-view/issues/666 – i still cannot delete files or folders from within the tree-view when the files are located on a mapped virtual drive.

Steps to reproduce:
- Create a new registry entry (and restart Windows afterwards):

```
REGEDIT4

[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Session Manager\DOS Devices] 
"W:"="\\??\\D:\\htdocs"
```

_edit:_
`subst W: D:\htdocs` – causes the same problem
- Create a file on your freshly mapped DOS device `W:\example\example.txt`
- Open folder `W:\` in atom
- Try to delete `W:\example\example.txt` using the tree-view -> fails

Opening `D:\htdocs` instead and then trying to delete `example.txt` works fine. File and folder rights are identical. Creating files or folders or renaming them works completely fine, only deletion is not possible (see my attached screenshots).

Comparison:
![comparison](http://i.imgur.com/E6vDk31.png)

Broken
![broken](http://i.imgur.com/kzzPTLe.png)

Deletion when opening `D:\htdocs` instead works fine!

**Comment by Ben3eeE** [comment]
I have the same issue with network mapped drives on Windows. Always assumed it was a duplicate of https://github.com/atom/tree-view/issues/830 so never created an issue for it.

**Comment by manuelbieh** [comment]
Seems to be pretty similar, indeed.

**Comment by manuelbieh** [comment]
I think I found the reason: virtual drives do not have an explicit recycle bin folder where Atom could move the files into:

```
Manuel@Manuel-406 /cygdrive
$ ls /cygdrive/c/'$Recycle.Bin'
S-1-5-18  S-1-5-21-2028212831-3322830245-2971025389-1001  S-1-5-21-4197599123-4253082845-4033610845-500

Manuel@Manuel-406 /cygdrive
$ ls /cygdrive/w/'$Recycle.Bin'
ls: Zugriff auf '/cygdrive/w/$Recycle.Bin' nicht möglich: No such file or directory
```

That's probably the reason why it fails: the folder Atom is trying to move the file into just does not exist.

**Comment by manuelbieh** [comment]
this workaround by saeed sagharichi works for me at the moment!

http://answers.microsoft.com/en-us/windows/forum/windows_7-files/windows-7-recycle-mapped-drives/95751679-428c-44b0-a8db-1783df9b5e13#3ca21578-3472-443c-8037-bfbdac1dd398

**Comment by 50Wliu** [comment]
Blocked on electron/electron#4974.

**Comment by Ben3eeE** [comment]
@50Wliu https://github.com/atom/tree-view/issues/830 was mac only and closed for this Windows specific. It appears to be the same problem on both platforms, deleting from a virtual drive without a recycle bin however. So maybe this and the electron issue should change labels?

**Comment by yassersharaf** [comment]
Windows 10 X64 still can't delete files to trash while recycle bin is enabled and i'm deleting project files from D:\XAMPP\htdocs folder
Any working solution?

**Comment by 50Wliu** [comment]
Your `D:\` directory most likely doesn't have a Recycle Bin.

**Comment by Ben3eeE** [comment]
Also very similar but on Linux https://github.com/atom/tree-view/issues/345#issuecomment-135779498

**Comment by jcgertig** [comment]
I cant add a recycling bin on my drive because it is a vdi any work arround for delete

**Comment by slikts** [comment]
@manuelbieh Thank you, the Windows workaround works.

**Comment by mjmfighter** [comment]
Has there been any updates to this recently?  Is there no way to not have something delete without a Recycle bin?  Like a Permanent delete instead of just moving it to a Recycle bin?  Would be nice to be able to delete things from mounted drives without having to do any extra workaround steps

**Comment by slikts** [comment]
A permanent delete without confirmation would be too dangerous.

**Comment by mjmfighter** [comment]
@slikts I didn't say without confirmation.  You can still have a prompt that says to confirm this deletion, but when I try to delete something right now all I get is a "Move to the Recycle bin" button.  This is why it was causing issues is because it was trying to move it to the Recycle bin, which didn't exist on that drive.  Having another button on there to permanently delete an item would be nice, even if there was another prompt of "are you really sure"

I just want to be able to delete extra folders or files I make on accident without having to open up explorer to delete them

**Comment by TeapotNickS** [comment]
any news on this bug for mac?

**Comment by 50Wliu** [comment]
@TeapotNickS This is a Windows-only bug.  If you are experiencing trouble deleting files on macOS, please open a new issue.

**Comment by TeapotNickS** [comment]
@50Wliu thanks ill open a new issue.

**Comment by Nektro** [comment]
I have a project folder I hold on a flash drive on Windows so that I can use atom on multiple computers, but this bug prevents me from deleting anything. Any word on this?

**Comment by KSXGitHub** [comment]
I have an idea: At first, try to move item to trash (using electron API), if fail, ask user to delete file permanetly (using nodejs API). This should prevent any issues related to electron's `shell.moveItemToTrash` in the future.

```coffeescript
if not shell.moveItemToTrash item
  if askUserToDeletePermanently()
    fs.unlink item
```

**Comment by BlaM** [comment]
So far I was using the addon package https://github.com/olmokramer/atom-permanent-delete to work around this issue, but the addon now has its own bug, so now I have to leave Atom every time I need to delete a file or folder. Any plans to get this fixed?

**Comment by oizik** [comment]
Just hit the exact same problem, as I just came from Linux onto a Windows machine for development. 

The solution that worked for me: [https://social.technet.microsoft.com/Forums/en-US/a349801f-398f-4139-8e8b-b0a92f599e2b/enable-recycle-bin-on-mapped-network-drives](https://social.technet.microsoft.com/Forums/en-US/a349801f-398f-4139-8e8b-b0a92f599e2b/enable-recycle-bin-on-mapped-network-drives)

I am running on a Windows 10 Pro Version 1703

**Comment by NxxN39** [comment]
Try re-indexing your drives or turn off drive indexing.

**Comment by afliw** [comment]
Are there currently any plans on fixing this issue?
I have all recycle bins on my disks disabled on Windows 10, and I don't want them enabled.
It's a real PITA have to leave Atom just to delete a file.

Please, fix it guys.
Thanks.

**Comment by 50Wliu** [comment]
We are currently waiting for the associated Electron issue to be resolved.

**Comment by dark-swordsman** [comment]
@oizik I tried all the steps in that and wasn't able to fix the issue. Also on Win 10 Ver 1703. 

Also, what do you guys mean by "enable the recycling bin"? I've gotten as far as showing the $Recycling.Bin folder and making sure it can be written to, but Atom still doesn't work with it. Very confused.

**Comment by afliw** [comment]
This is what I mean by _"recycle bin disabled"_:
![image](https://user-images.githubusercontent.com/6609070/30290787-9a7cad56-9728-11e7-8065-b55cc12851d5.png)

I'm currently using the package [permanent-delete](https://atom.io/packages/permanent-delete) with the fix for it found [here](https://github.com/olmokramer/atom-permanent-delete/issues/10) by user @nexter-av as a workaround. Works great at the moment.

**Comment by DannyJJK** [comment]
This isn't a Windows-only issue, it happens anywhere where there is no concept of a "Trash" or "Recycle Bin", e.g. using SSHFS on my Mac to mount a Linux drive, results in the same error message when trying to delete a file on there.

Why can't we do what VS Code does, and have it say "This file cannot be moved to trash, do you want to delete permanently?" then give the option to the user. On VS Code I just do this, but on Atom every time I want to delete a file I have to open the file in Finder/Explorer then delete it through there, which isn't a very user-friendly experience

**Comment by timpfairsailcom** [comment]
Come on guys, this thread began more than 16 months ago.  Still no fix?

**Comment by geoffryb** [comment]
I had a similar issue.

Couldn't delete folders on my Storage/Workfolder (F) drive. I noticed that the recycle bin was called $RECYCLE.BIN on F and $Recycle.Bin on C.

Changed the name of the F folder to $Recycle.Bin and it worked.

**Comment by dark-swordsman** [comment]
@DannyJJK > Why can't we do what VS Code does, and have it say "This file cannot be moved to trash, do you want to delete permanently?" then give the option to the user.

I completely agree with that. I was able to enable recycle bin and it works now, but that should, by default, be a feature.

**Comment by AlicanGul** [comment]
I don't use Recycle Bin. I'm sure many people don't. Really don't get why can't atom delete files permanently.

**Comment by davidsusu** [comment]
I am on Debian, with disabled trash. I have tried Atom today for the first time on this machine, and I almost uninstalled because I could not delete any file. Fortunately, the `permanent-delete` plugin helped. However, at least `Shift+Del` or something similar should be work **by default**.

**Comment by BlaM** [comment]
Is there an option to disable the built-in "delete" functionality completely. It's sometimes confusing to have two delete options with one of them not working.

**Comment by lee-dohm** [comment]
Thanks for the feedback.

This is a specific design decision. If Atom can't find the Recycle Bin, Trash or whatever OS analog there is, Atom will refuse to delete a file. We have simply had too many people running into problems with accidentally deleting files they didn't intend to delete and blaming Atom. It requires too much support effort for something that is a nice-to-have. If people would like to have the error message improved so that it provides more context, that is definitely something that we would evaluate.

Since this issue, as it is formulated, isn't something that we're going to correct. I'm going to close this. Thanks again for all your input.

**Comment by Cylindric** [comment]
Yeah, why behave like every other app in existence and follow standard OS conventions? Horrible decision
 :(

**Comment by samuelharmer** [comment]
> We have simply had too many people running into problems with accidentally deleting files they didn't intend to delete and blaming Atom

[Idiocracy](http://www.imdb.com/title/tt0387808/) is almost upon us.

**Comment by alexus85** [comment]
well our security rules disabled trash across all computers ... So what about making an options somewhere in the settings which would override the default behavior? .. Meaning that the file would be deleted permanently when such option is enabled.  

If that's not possible, I suppose I have no other choice but switch to some other IDE.

**Comment by 50Wliu** [comment]
@alexus85 you can install packages such as [permanent-delete](https://atom.io/packages/permanent-delete) if you wish to do so.

**Comment by bitsnaps** [comment]
I believe `Shift + Del` is most accurate choice, since this is the most used behavior for many OS not just Windows, I like the default behavior for `Del` key so far.

**Comment by lucas-pouchot** [comment]
> We have simply had too many people running into problems with accidentally deleting files they didn't intend to delete and blaming Atom

Peoples who lose files like this shouldn't blame Atom but start using a versioning tool like git.
having to delete his files outside the editor is penalizing :(

**Comment by vmvwebworks** [comment]
someone checked this? https://askubuntu.com/questions/288513/cant-move-files-to-the-trash

**Comment by skaraman** [comment]
I don't use a virtual or mapped drive yet I have this issue. I AM running on bootcamp but only have the issue on one bootcamp device out of multiple. 

Please consider allowing setting a 'delete' destination as everyone should be aware that even the Recycle Bin is not deleting any files but simply storing them in a place yet to be permanently deleted. How can anyone blame Atom for possibly moving files to the RB if the user has to manually purge the RB themselves? Recycle Bin is just a folder.

**Comment by manngo** [comment]
> Since this issue, as it is formulated, isn't something that we're going to correct. I'm going to close this. Thanks again for all your input.

Sorry, but I don’t think that’s a very realistic approach. Anybody who blames a text editor for deleting files when requested has missed the point, and normal users who know what they’re doing shouldn’t be punished for their ignorance.

This isn’t just nice to have. There are many reasons that a volume doesn’t implement a Bin, and Atom shouldn’t make the arbitrary decision to imagine that it does. That’s not the way software should work.

**Comment by xorinzor** [comment]
Can't believe this issue is basically closed as a "wont-fix".

I guess I will just have to abandon Atom then, as it's very annoying that I can't delete files within Atom without applying some odd workaround of creating a fake recycle-bin. It's a network share, it doesn't have nor need a recycle-bin.

**Comment by Cylindric** [comment]
@xorinzor another reason I switched to VSCode. I doubt anyone is even reading these Closed issues.

**Comment by dark-swordsman** [comment]
Yeah, since I made my comment over a year ago, I've just been using VSCode. Much better extensions and runs better AFAIK.

Haven't really had any issues with VSCode on Ubuntu 18.04 LTS. 🤷‍♂️

**Comment by DannyJJK** [comment]
It’s funny how the above comments were marked as off-topic. VS Code has worked fine without a recycle bin since I started using it, possibly since its first release. Making a good Electron app is to make it so that it doesn’t feel like you’re using Electron or something independent of how the OS works. If deleting a file without a recycle bin is possible outside of Atom, it should be possible inside of Atom, just give the usual confirm dialog of “Are you sure you want to delete this?” with a message of how there’s no recycle bin option. It’s really not that hard, and ignoring the issue or hiding comments is really just showing why VS Code has taken over. I’d love to love Atom, but ignoring the user base is the number 1 reason as to why you lose users

**Comment by marcpeterson** [comment]
If you're seeing the error "The following file couldn’t be moved to trash” and “Is there a Recycle Bin on the drive where the files are stored”, then your D drive's recycle bin alias may be named differently than in your C drive.  Seems like Atom cannot handle that.

I was able to fix this in Windows 10.  The recycle bin in your C drive is a hidden folder called $Recycle.Bin.  In your D drive it may be called $RECYCLE.BIN.  Renaming the D drive alias to match the C drive folder fixed this.  I do not know if this will cause other side effects.  Although Windows is supposed to be case-insensitive, so hopefully not.

To view a drive's hidden files, in Explorer go to View -> Options -> Folder Options -> View -> Advanced Settings and uncheck "Hide protected operating system files".

**Comment by MichaelJCole** [comment]
This bug also affects files loaded using WSL2 paths.  Permanent delete package doesn't work.  Ok, fine.  VSCode it is.

**Comment by GCSBOSS** [comment]
Why you keep pushing me to VSCode. It's like you don't want me anymore 😭
Seriously, is there any way Atom can not be unbroken for a lot of users o this matter?

**Comment by msbeck** [comment]
Just adding my two cents. This still happens on Mac Big Sur 11.1

Developing code to run ON a usb attached device and need to move files in and out often. Workflow is edit a code file on the local drive, and send it off to run on the usb device. Don't want to consistently work out of a non stable drive and have to constantly keep backups, just send code as necessary to run to the device when needed.

Cannot delete within the editor. Would like to be able to drag from one directory over to the device to overwrite files as necessary, but the filenames are appended with 0, etc. so cannot replace. Have to open up finder and delete a file I need replaced, or go through the save dialog, but that switches my future edits to ctrl+s save onto the target device rather than the local version.

**Comment by manngo** [comment]
> @alexus85 you can install packages such as [permanent-delete](https://atom.io/packages/permanent-delete) if you wish to do so.

I don’t know why this was downvoted. It doesn’t solve the fundamental problem that some Atom designer made a weird decision about usability and chooses not to listen to users who want it fixed, but it does at least offer a workaround.

I myself have found this workaround very useful.

**Comment by wKovacs** [comment]
This has still no fix?

The easiest solution would be: inform the user there is no recycle bin AND offer the option to delete permanently in form of easy yes or no question.

right now I use permanent-delete. Most of the time I go for "normal" delete just out of habit and the error message is annoying in addition I have to issue another comment instead of getting a solution offered

**Comment by darrenthebozz** [comment]
This is really annoying on a domain that prevents such tasks as having a recycling bin.
